### PR TITLE
feat: Add promotion support for pipelines

### PR DESCRIPTION
**Pull request by BYK**
Adds support for *automatically* detecting an app in a pipeline and use promotion when possible.

### CODE REVIEW

**Review by micheletest — CHANGES_REQUESTED** [review]
Can you add some documentation to this:

- This seems to add a few new outputs so specify that in the readme
- The pipeline check step is more complicated than anything previous. It's a bit hard to follow. I think it could benefit from some comments on what it's doing, either in the readme or the step itself.

**Comment by BYK** [comment]
>This seems to add a few new outputs so specify that in the readme

These are internal outputs used by other steps, not exposed to any consumer so nothing to document there :)

**Review by micheletest — APPROVED** [review]
